pkg/bmc/idrac_virtualmedia.go

```diff
@@ -90,3 +92,52 @@ func (a *redfishiDracVirtualMediaAccessDetails) VendorInterface() string {
 func (a *redfishiDracVirtualMediaAccessDetails) SupportsSecureBoot() bool {
 	return true
 }
+
+func (a *redfishiDracVirtualMediaAccessDetails) ValidateBootInterface(bmcCreds Credentials) (bool, error) {
+	bmcConfig := gofish.ClientConfig{
+		Endpoint: getRedfishAddress(a.bmcType, a.host),
+		Username: bmcCreds.Username,
+		Password: bmcCreds.Password,
+		Insecure: a.disableCertificateVerification,
+	}
+
+	client, err := gofish.Connect(bmcConfig)
+	if err != nil {
+		return false, err
```

Comment:
We should wrap all of the errors in this function to provide more context about what was going on.

Reply:
Ack, I will take a look on how to do this =)

Reply:
The function you want is `errors.Wrap()` from `github.com/pkg/errors`. There are some other examples in other parts of this code base.

Reply:
Thanks!

Reply:
Done